app/controllers/api/functions.php

```diff
@@ -160,6 +169,8 @@
     ->param('templateOwner', '', new Text(128, 0), 'The name of the owner of the template.', true)
     ->param('templateRootDirectory', '', new Text(128, 0), 'Path to function code in the template repo.', true)
     ->param('templateBranch', '', new Text(128, 0), 'Production branch for the repo linked to the function template.', true)
+    ->param('memory', 512, fn (array $plan) => empty($plan['memoryLimit']) ? new WhiteList($filterBelowThreshold(MEMORY_VALUES, System::getEnv('_APP_FUNCTIONS_MEMORY')), false, WhiteList::TYPE_INTEGER) : new WhiteList($filterBelowThreshold(MEMORY_VALUES, $plan['memoryLimit']), false, WhiteList::TYPE_INTEGER), 'Memory in MB allocated for the function.', true, ['plan'])
```

Comment:
I dont think $plan will be here,  it's Cloud-only.
I would suggest special Memory and CPU validators here instead. Then, on Cloud, we can override those validators and add billing logic.

Reply:
This comment applies to all places where you used $plan

Reply:
Created the new validators, please re-review